theano/scalar/basic.py

```diff
@@ -3099,7 +3109,8 @@ def init_name(self):
                 for i, r in enumerate(self.fgraph.variables):
                     if r not in io and len(r.clients) > 1:
                         r.name = 't%i' % i
-                rval = "Composite{%s}" % str(self.fgraph)
+                rval = "Composite{%s}" % ', '.join([pprint(output) for output
+                                                    in self.fgraph.outputs])
```

Comment:
I didn't look at the output, maybe it make it ugly, so I let you try it and decide what to do. If you do not do it, just put in a comment the output that you find ugly.